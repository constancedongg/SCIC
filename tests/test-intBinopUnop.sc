int func main() {
    print(10 + 19801);          /* Add */ 
    print(0 - 1234567);         /* Sub */ 
    print(100 * 900000);        /* Mul */ 
    print(90 / 80);             /* Div -> floor */ 
    print(-90 / 80);            /* Div -> sign abs(floor(res)) */
    printb(2  > -2);            /* Greater */
    printb(1 + 4 == 5);         /* Equal, Add */
    printb(7 == 30 / 4);       /* Equal, Div */
    printb(12345678901234 >= 12345678901233);      /* Geq */
    print(2147483647);
    print(2147483648);
    printb(2147483647 > 2147483648);      /* Greater */
    printb(-100 <= 51-151);       /* Leq */
    printb(-91 < -1);             /* Less */
    print(-187);                    /* Neg */
    return 0;
}