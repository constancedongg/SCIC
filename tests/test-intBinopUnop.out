19811
-1234567
90000000
1
-1
1
1
1
1
2147483647
-2147483648
1
1
1
-187